.devcontainer/Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM python:3.11.5-slim-bookworm
+# -----------------------------------------
+# We use the "stable" debian release; python does not seem to release current
+# versions on LTS debian releases.
+
+# We don't upgrade, and we pin all os dependency versions
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends zip=3.0-13 unzip=6.0-28 bzip2=1.0.8-5+b1 libmagic-dev=1:5.44-3 htop=3.2.2-2 wget=1.21.3-1+b2 git=1:2.39.2-1.1 openssh-client=1:9.2p1-2 && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+SHELL ["/bin/bash", "-c"]
+
+RUN mkdir -p /app
+
+WORKDIR /app
+
+# Standard simplified python setup.
+COPY ./requirements.txt /app/requirements.txt
```

Comment:
The existing documentation says that the requirements.txt file is the one that is used, and that the Pipfile is there and (possibly) maintained for those who like to use it for development.  
Personally, I have never seen a backup package manager in a repo like this, and wish it weren't so. In some JS codebases we might have had npm and bower in the same repo, but that was a transition from an old package manager to a newer one.
I'd be ok with Pipfile, though I think if we switch the package manager Poetry would be a better choice than pipenv.
IMO It would be nice for KBase to support just 2 Python package managers - plain pip with requirements.txt (since it is so common) for legacy projects (and it is the one supported by kb-sdk), and Poetry for new repos.